### PR TITLE
Constraint to enforce stackdriver logging on a GKE cluster

### PR DESCRIPTION
**Pull request by red2k18**

### CODE REVIEW

File: samples/gke_enable_stackdriver_logging.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+apiVersion: constraints.gatekeeper.sh/v1alpha1
+kind: GCPGKEEnableStackdriverLoggingConstraintV1
+metadata:
+  name: enable_gke_stackdriver_logging
+  annotations:
+    description: Ensure stackdriver logging is enabled on a GKE cluster
+    benchmark: CIS11_7.02
```
Comment (joecheuk) [inline samples/gke_enable_stackdriver_logging.yaml]: Can you update this to the latest format: https://github.com/forseti-security/policy-library/blob/ec9b5c2040d9afe2a551d8b78505eb97a55d376a/samples/gke_node_pool_auto_repair.yaml#L22

Please also make sure this is CIS 7.02

**Comment by red2k18** [comment]
Updated

File: validator/test/fixtures/gke_enable_stackdriver_logging/constraints/enable_gke_stackdriver_logging/data.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+apiVersion: constraints.gatekeeper.sh/v1alpha1
+kind: GCPGKEEnableStackdriverLoggingConstraintV1
+metadata:
+  name: enable_gke_stackdriver_logging
+  annotations:
+    description: Ensure stackdriver logging is enabled on a GKE cluster
+    benchmark: CIS11_7.02
```
Comment (joecheuk) [inline validator/test/fixtures/gke_enable_stackdriver_logging/constraints/enable_gke_stackdriver_logging/data.yaml]: You can remove the benchmark in the test file

File: validator/gke_enable_stackdriver_logging_test.rego [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+test_stackdriver_logging_enabled {
+	violation := all_violations[_]
+	resource_names := {x | x = all_violations[_].details.resource}
+	not resource_names["//container.googleapis.com/projects/transfer-repos/zones/us-central1-c/clusters/joe-clust"]
+}
+
+test_stackdriver_logging_disabled {
+	violation := all_violations[_]
+	violation.details.resource == "//container.googleapis.com/projects/transfer-repos/zones/us-central1-c/clusters/joe-clust2"
```
Comment (joecheuk) [inline validator/gke_enable_stackdriver_logging_test.rego]: joe-clust2 also has loggingservice enabled according to the test data.

"loggingService": "logging.googleapis.com",

this should not be part of this test method - have you tried running the test cases?

**Comment by red2k18** [comment]
@joecheuk thanks for the review. Addressed all comments. Tests are working now.

File: validator/gke_enable_stackdriver_logging_test.rego [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	resource := data.test.fixtures.gke_enable_stackdriver_logging.assets[_]
+	constraint := data.test.fixtures.gke_enable_stackdriver_logging.constraints.enable_gke_stackdriver_logging
+
+	issues := deny with input.asset as resource
+		 with input.constraint as constraint
+
+	violation := issues[_]
+}
+
+test_stackdriver_logging_enabled {
```
Comment (joecheuk) [inline validator/gke_enable_stackdriver_logging_test.rego]: Can you update the method name to describe the scenario you are testing on?

**Pull request merged**